.github/workflows/workflow.yml

```diff
@@ -1,20 +1,161 @@
-# This is a minimal workflow.
-
-name: CI
 
+name: Main workflow
 on:
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+  ## No sense in a dev building every time they push and no one
+  ## should be working on the common branches.
+  push:
+    branches:
+      - master
+      - develop
+      - 'hotfix/**'
+      - 'release/**'
+      - 'feature/**'
+  ## Any pull request. Yes the syntax looks weird
+  pull_request:
+
 jobs:
 
-  build:
 
+  test:
+    name: Test the loader on ${{matrix.operating-system}}
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macOS-latest]
+
+    steps:
+      - name: Get the code
+        uses: actions/checkout@v2
+
+      - name: Set up node
+        uses:  actions/setup-node@v3
+        with:
+          node-version: 16
```

Comment:
Please look at the cgov-react-app-playground for a [better setup node](https://github.com/NCIOCPL/cgov-react-app-playground/blob/develop/.github/workflows/workflow.yml#L64) It will cache npm packages and uses the nvmrc file to set the node version. (So we don't need to touch the workflow when going to a new version of node)

Reply:
Done.